src/image_copy/main.go

```diff
@@ -69,44 +73,60 @@ func getTags(dstTag string, updateMajor bool) []string {
 	return tags
 }
 
+func execCmd(name string, args ...string) string {
+	cmd := exec.Command(name, args...)
+	fmt.Println("\n> ", name, strings.Join(args, " "))
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		fmt.Println(string(out))
+		log.Fatal(err.Error())
+	}
+	return string(out)
+}
+
 func mustHaveArchitecture(imageTag, platform, expectedArch string) {
-	fmt.Printf("verifying binaries in '%s' from '%s' for platform '%s' are '%s'\n", binFolder, imageTag, platform, expectedArch)
+	fmt.Printf("\nverifying binaries in %q from %q for platform %q are %q\n", binFolder, imageTag, platform, expectedArch)
 
-	cmd := exec.Command(
+	// create local container from image
+	containerId := base64.RawURLEncoding.EncodeToString([]byte(fmt.Sprintf("%v-%v", platform, time.Now().UnixMilli())))
+	out := execCmd(
 		"docker",
-		"run",
+		"create",
+		"--name", containerId,
 		"--platform", platform,
-		"--user=root",
-		"--rm",
-		"--entrypoint=sh",
-		imageTag,
-		"-c", `
-apk add -U file > /dev/null
-
-count=0
-for file in `+binFolder+`/*; do
-  if [ -f "$file" ]; then
-    count=$((count+1))
-    if file "$file" | grep -q "`+expectedArch+`"; then
-      echo "$file is `+expectedArch+`"
-    else
-      echo "$file is not `+expectedArch+`"
-      file "$file"
-      exit 1
-    fi
-  fi
-done
-
-if [ $count -lt 1 ]; then
-  echo "no binaries were found"
-  exit 1
-fi
-`)
-	out, err := cmd.CombinedOutput()
-	fmt.Println(string(out))
+		imageTag)
+	fmt.Println(out)
+	defer execCmd("docker", "rm", containerId)
+
+	// copy binaries to local tmp dir
+	tmpDir, err := os.MkdirTemp(os.TempDir(), containerId)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+	out = execCmd(
+		"docker",
+		"cp",
+		containerId+":"+binFolder+"/.",
+		tmpDir)
+	fmt.Println(out)
+
+	// verify binaries
+	files, err := ioutil.ReadDir(tmpDir)
```

Comment:
```suggestion
	files, err := os.ReadDir(tmpDir)
```
ioutil is deprecated

Reply:
Oddly enough, Goland showed me a warning for the other functions but not this one.